crypto/http/http_client.c

```diff
@@ -135,7 +135,7 @@ void OSSL_HTTP_REQ_CTX_set_max_response_length(OSSL_HTTP_REQ_CTX *rctx,
         ERR_raise(ERR_LIB_HTTP, ERR_R_PASSED_NULL_PARAMETER);
         return;
     }
-    rctx->max_resp_len = len != 0 ? len : HTTP_DEFAULT_MAX_RESP_LEN;
+    rctx->max_resp_len = len != 0 ? (size_t)len : HTTP_DEFAULT_MAX_RESP_LEN;
```

Comment:
I suppose the signature of SSL_HTTP_REQ_CTX_set_max_response_length should be changed to size_t as well?

Reply:
I suppose we need to keep the type of `OSSL_HTTP_REQ_CTX_set_max_response_length()` for backward compatibility, so deliberately did not update that.

Reply:
Ah, right, it is used by the old OCSP call macro. That could be made a deprecated wrapper function instead.

Reply:
Not sure it is worth it though.

Reply:
I don't this it's worth the effort.
People are anyway better off using the high-level HTTP API,
and the difference between the old `unsigned long` and the nicer `size_t` should be negligible in practice.